pages/_app.js

```diff
@@ -1,109 +1,116 @@
 import Head from 'next/head'
 import fetch from 'isomorphic-unfetch'
-import { useRef, useState, useEffect } from 'react'
-import getConfig from 'next/config'
+import { useState, useEffect } from 'react'
 import {useRouter} from 'next/router'
-import AppContext from './AppContext'
 import CountrySelector from './Components/CountrySelector'
+import rootReducer from './reducers'
+import { createStore } from 'redux'
+import { Provider, useSelector, useDispatch } from 'react-redux'
+import { receiveCountries, selectCountry, selectYear, fetchCountries } from './actions'
 
-function App({ Component, pageProps }) {
+const App = () => {
 
+  const dispatch = useDispatch()
   const router = useRouter()
 
-  const [countries, setCountries] = useState([])
-  const [country, setCountry] = useState("")
-  const [year, setYear] = useState("")
+  const country = useSelector((state) => state.country)
+  const year = useSelector((state) => state.year)
+  const countries = useSelector((state) => state.countries)
+
   const [years, setYears] = useState([new Date().getFullYear() - 1])
 
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    router.push(`/holidays?country=${country}&year=${year}`)
+  }
+
+  const changeCountry = value => {
+    dispatch(selectCountry(value))
+  }
+
+  const changeYear = value => {
+    dispatch(selectYear(value))
+  }
+
   if(!country && router.query.country) {
-    setCountry(router.query.country)
+    changeCountry(router.query.country)
   }
   if(!year && router.query.year) {
-    setYear(router.query.year)
+    changeYear(router.query.year)
   }
 
-  // To fetch .env variables from next.config.js
-  const {publicRuntimeConfig} = getConfig()
-  const {holidayApiURL, holidayApiKey} = publicRuntimeConfig
-
   useEffect(() => {
```

Comment:
Code written in `useEffect` should not be a part of a component. it should be part of a actions